dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
```

Comment:
There is couple reason behind that:
1- the bug about getDate only concern message built by MessageBuilder and there is other way to build message than this messageBuilder. Message getDate method is in fact not buggy and respect the doc. This is MessageBuilder which is buggy. That why I choose to not modify the Message class like I said I will do on Mattermost
2 - Yes I could have done a .notOverrideNullDate I thought about that but I choose to not do that. Firstly because that would be ugly, secondtly because I wanted to express user of the lib that the MessageBuilder is in this current state buggy by the intermediate of @Deprecated warning ! If I add the .notOverrideNullDate(), I would not be able to mark current way to build the MessageBuilder deprecatted and most of the user of this library will never know about the current issue and will never check theire code and call this `notOverrideNullDate()`.
Lastly when we will be able to move to next major version and break the API then the `.notOverrideNullDate()` method would have no sens anymore and would be removed so this would be a very ephemeral method.
With my solution moving to next major version would be more easy and more logic, we would just remove all the deprecated constructor method. 

I have the impression you see deprecation as a unwanted thing, where I see it as a strenghnest of my PR to alert user. And yes my PR is more work than adding a `.notOverrideNullDate()` but I think it definitly worst it.